dev/integration_tests/deferred_components_test/run_release_test.sh

```diff
@@ -32,15 +33,23 @@ java -jar $bundletool_jar_path install-apks --apks=build/app/outputs/bundle/rele
 
 $adb_path shell "
 am start -n io.flutter.integration.deferred_components_test/.MainActivity
-sleep 30
 exit
 "
-$adb_path logcat -d -t "$script_start_time" > build/app/outputs/bundle/release/run_logcat.log
-echo ""
-if cat build/app/outputs/bundle/release/run_logcat.log | grep -q "Running deferred code"; then
-  echo "All tests passed."
-  exit 0
-fi
-cat build/app/outputs/bundle/release/run_logcat.log
+while read LOGLINE
+do
+    if [[ "${LOGLINE}" == *"Running deferred code"* ]]; then
+        echo "Found ${LOGLINE}"
+        pkill -P $$
+        echo "All tests passed."
+        exit 0
+    fi
+    # add timeout
+    current_time=$(date +%s)
+    if [[ $(( $current_time - $script_start_time_seconds )) -ge 150 ]]; then
```

Comment:
nit: You don't need the '$' inside of "$((" and "))". You also don't need the extra spaces inside the double-parens (although it's fine to leave them).
```suggestion
    if [[ $((current_time - script_start_time_seconds)) -ge 150 ]]; then
```